src/Config.php

```diff
@@ -1233,7 +1250,21 @@ public function processLongArgument($arg, $pos)
                     break;
                 }
 
-                $this->generator = substr($arg, 10);
+                $generatorName          = substr($arg, 10);
+                $lowerCaseGeneratorName = strtolower($generatorName);
+
+                if (isset(self::$validGenerators[$lowerCaseGeneratorName]) === false) {
+                    $validOptions = implode(', ', array_values(self::$validGenerators));
```

Comment:
Is the `array_values()` really needed here ?

Reply:
Also wondering if for readability, the last comma should be replaced with " and " ?

Reply:
> Is the array_values() really needed here ?

Ops, no. It was added by mistake.

> Also wondering if for readability, the last comma should be replaced with " and " ?

Yes, I think it is slightly better, and I used it in the original version. But then, when I moved to store the list of valid generators in a class property and used `implode()` to get the string with the name of the generators, I removed "and", assuming that the text is good enough without it, still correct and the code is slightly simpler.

As you are asking, I'm assuming you prefer the version with the "and," so I went ahead and added it. I'm not super happy with my approach using `array_pop()`, but I couldn't think of anything better. Please let me know if you have another suggestion.

Using `array_pop()` meant that I had to change `Config::$validGenerators` to an object property instead of a class property which is related to another comment you left.

Reply:
Ah, but this is supposed to be an unchangeable array (to become a class constant later on as discussed in previous comments) and the use of `array_pop()` explicitly changes the array (by reference) and would block the future change to make this a class constant. See: https://3v4l.org/QrM61

While the change of the property value would not currently be problematic (as a second `--generator=...` argument is ignored anyway), it does make the code less stable, so I'd like to see an alternative solution for this.